MatrixSDK/Data/MXRoomLastMessage.m

```diff
@@ -87,16 +104,39 @@ - (instancetype)initWithManagedObject:(MXRoomLastMessageMO *)model
         _originServerTs = model.s_originServerTs;
         _isEncrypted = model.s_isEncrypted;
         _sender = model.s_sender;
-        _text = model.s_text;
-        if (model.s_attributedText)
+        
+        NSData* archivedSensitiveData;
+        if (model.s_sensitiveData && model.s_isEncrypted)
         {
-            _attributedText = [NSKeyedUnarchiver unarchiveObjectWithData:model.s_attributedText];
+            archivedSensitiveData = [self decrypt:model.s_sensitiveData];
         }
-        if (model.s_others)
+        else
         {
-            _others = [NSKeyedUnarchiver unarchiveObjectWithData:model.s_others];
+            archivedSensitiveData = model.s_sensitiveData;
+        }
+        
+        if (archivedSensitiveData)
+        {
+            NSDictionary* sensitiveDataDictionary = [NSKeyedUnarchiver unarchiveObjectWithData:archivedSensitiveData];
+            
+            _text = sensitiveDataDictionary[kCodingKeyText];
+            _attributedText = sensitiveDataDictionary[kCodingKeyAttributedText];
+            _others = sensitiveDataDictionary[kCodingKeyOthers];
+        }
+        else // fallback logic for old database versions
```

Comment:
You can completely avoid having to deal with old versions by incrementing `kMXFileVersion` in `MXFileStore`. It'll wipe the data and force a fresh sync when the user upgrades. Probably better than running the old way in parallel and avoids having to think about migrations etc.

Reply:
Technically I didn't change the `MXFileStore` so I'm not sure this is clean.
Btw does it mean people will need to do a full sync potentially taking a lot of time (for account big as Matthew's one)?